beacon-chain/p2p/service.go

```diff
@@ -551,3 +572,27 @@ func (s *Service) downscorePeer(peerID peer.ID, reason string) {
 	newScore := s.Peers().Scorers().BadResponsesScorer().Increment(peerID)
 	log.WithFields(logrus.Fields{"peerID": peerID, "reason": reason, "newScore": newScore}).Debug("Downscore peer")
 }
+
+func (s *Service) AttestationSubnets(nodeID enode.ID, node *enode.Node, record *enr.Record) (map[uint64]bool, error) {
+	if !s.filterPeer(node) {
```

Comment:
Should crawler also receive `filterPeer` as a dependency so we don't need to repeat this call in each extractor?